bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/StringUtils.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.runtime.internal.util;
+
+import io.ballerina.runtime.api.TypeTags;
+import io.ballerina.runtime.api.types.MethodType;
+import io.ballerina.runtime.api.types.ObjectType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BLink;
+import io.ballerina.runtime.api.values.BRefValue;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BValue;
+import io.ballerina.runtime.internal.BalStringUtils;
+import io.ballerina.runtime.internal.CycleUtils;
+import io.ballerina.runtime.internal.TypeChecker;
+import io.ballerina.runtime.internal.regexp.RegExpFactory;
+import io.ballerina.runtime.internal.scheduling.Scheduler;
+import io.ballerina.runtime.internal.util.exceptions.BallerinaException;
+import io.ballerina.runtime.internal.values.AbstractObjectValue;
+import io.ballerina.runtime.internal.values.ArrayValue;
+import io.ballerina.runtime.internal.values.DecimalValue;
+import io.ballerina.runtime.internal.values.MapValueImpl;
+import io.ballerina.runtime.internal.values.ObjectValue;
+
+/**
+ * Common utility methods used for String manipulation.
+ *
+ * @since 2201.6.0
+ */
+public class StringUtils {
+
+    public static final String STR_CYCLE = "...";
+    public static final String TO_STRING = "toString";
+
+    private StringUtils() {}
+
+    /**
+     * Returns the human-readable string value of Ballerina values.
+     *
+     * @param value     The value on which the function is invoked
+     * @param parent    The link to the parent node
+     * @return          String value of the value
+     */
+    public static String getStringVal(Object value, BLink parent) {
+        if (value == null) {
+            return "";
+        }
+
+        Type type = TypeUtils.getReferredType(TypeChecker.getType(value));
+
+        if (type.getTag() == TypeTags.STRING_TAG) {
+            return ((BString) value).getValue();
+        }
+
```

Comment:
Check for newlines in other places as well.

Reply:
Will remove the newlines as it is expected by the WSO2 code style practices.

Reply:
Fixed with c24a2fbd2b99c9dd6dc76d45237874379999be35.